src/App.tsx

```diff
@@ -1,7 +1,37 @@
-import React from 'react';
+import { Component, ReactNode } from 'react';
 
-export const App: React.FC = () => (
-  <div className="App">
-    <p className="App__message">The last pressed key is [Enter]</p>
-  </div>
-);
+export class App extends Component {
+  state: Readonly<{ key?: string }> = {
+    key: '',
+  };
+
+  componentDidMount() {
+    document.addEventListener('keyup', (event: KeyboardEvent) => {
+      this.setState({
+        key: event.key,
+      });
+    });
```

Comment:
you two times write this callback. Create separate method for it
```suggestion
    document.addEventListener('keyup', your_callback);
```